Update Cloudflare Pages deploy guide (#1318)
withastro/docs: src/pages/en/guides/deploy/cloudflare.md

    - **Framework preset**: `Astro`
    - **Build command:** `npm run build`
    - **Build output directory:** `dist`

>     - **Environment variables (advanced)**: By default, Cloudflare Pages uses `NODE_VERSION = 12.18.0`, however Astro [requires](/en/install/auto/#prerequisites) `14.18.0`, `v16.12.0`, or higher. If you are not using a `.nvmrc. or `.node-version` file to manually specify a version, add an entry with a **Variable name** of `NODE_VERSION` and a **Value** of `14.18.0`, `v16.12.0`, or higher.

Reviewer: ```suggestion
    - **Environment variables (advanced)**: By default, Cloudflare Pages uses `NODE_VERSION = 12.18.0`. However, Astro [requires](/en/install/auto/#prerequisites) `14.18.0`, `v16.12.0`, or higher. Use a `.nvmrc.` or `.node-version` file to manually specify a version, or add an environment variable with a **Variable name** of `NODE_VERSION` and a **Value** of `14.18.0`, `v16.12.0`, or higher.
```

>     - **Environment variables (advanced)**: By default, Cloudflare Pages uses `NODE_VERSION = 12.18.0`. However, Astro [requires](/en/install/auto/#prerequisites) `14.18.0`, `v16.12.0`, or higher. Use a `.nvmrc.` or `.node-version` file to manually specify a version, or add an environment variable with a **Variable name** of `NODE_VERSION` and a **Value** of `14.18.0`, `v16.12.0`, or higher.

Reviewer 2: There are a lot of options here (nvmrc, node-version, env variable, v14, v16…). How do we feel about reducing that a little?

[Node 14 will reach end-of-life in April](https://nodejs.org/en/about/releases/), so we can probably safely recommend most folks choose 16 unless they really need 14. I also don’t see a reason to document both `.node-version` and `.nvmrc`.

I’ve also tried to simplify the text a little. Let me know what you think!

```suggestion
    - **Environment variables (advanced)**: By default, Cloudflare Pages uses Node.js 12.18.0, but Astro [requires a higher version](/en/install/auto/#prerequisites). Add an environment variable with a **Variable name** of `NODE_VERSION` and a **Value** of `v16.12.0` or higher to tell Cloudflare to use a compatible Node version. Alternatively, add a `.nvmrc` file to your project to specify a Node version.
```

To get started, create a `/functions` directory at the root of your project. Writing your Functions files in this directory automatically generates a Worker with custom functionality at the predesignated routes. To learn more about writing Functions, see the [Pages Functions documentation](https://developers.cloudflare.com/pages/platform/functions/).

📚 Read more about [SSR in Astro](/en/guides/server-side-rendering/).

### Troubleshooting

If you're encountering errors, double-check the version of `node` you're using locally (`node -v`) matches the version you're specifying in the environment variable.

If the versions match and you're still having trouble, try using `v16.12.0` or later, and `directory` mode:
```ts title="astro.config.mjs" "directory"
export default defineConfig({
  adapter: cloudflare({ mode: "directory" }),

> });```

Reviewer 2: Looks like these backticks jumped up a line!

````suggestion
});
```

````
